src/web/lib/ElementRenderer.tsx

```diff
@@ -201,6 +202,25 @@ export const ElementRenderer = ({
 			);
 		case 'model.dotcomrendering.pageElements.EmbedBlockElement':
 			if (!element.safe) {
+				if (isMainMedia) {
+					return (
+						<Figure
+							isMainMedia={isMainMedia}
+							role={isLiveBlog ? 'inline' : element.role}
+							id={element.elementId}
+						>
+							<iframe
```

Comment:
+1 for breaking this out of `UnsafeBlockComponent` - we don't need the iframe resizer here.

But I feel that semantically, ElementRenderer should be rendering an element _component_. Just because that's what it does everywhere else.

I suppose you could have a new type of element, like a MainMediaEmbedBlockComponent, but ooof, that's long!

Seems like the list of different Embeds is growing. We've got `UnsafeEmbedBlockComponent`, `EmbedBlockComponent` and now , potentially, `MainMediaEmbedBlockComponent`. I think there's a case here to refactor this somehow. Either we look at splitting types higher up, in Frontend or we could bring the variations down into `EmbedBlockComponent`. Not sure what the right approach is but I think we've hit the point when a refactor would be a good idea.

Reply:
In this case I do not feel we need to split this out into it own component, but I sense the main reason is not to import `css` here and give the idea that we should be donig styling in this file (so I will spin it out).

Personally I feel like `ElementRenderer.tsx` could be split, one file for Main media rendering, the other for body content, as there are lots of components that would not make sense to be rendered as Main media (ex: SpotifyBlockElement, TimelineBlockElement or QuizAtomBlockElement to name a few)

Reply:
> Personally I feel like ElementRenderer.tsx could be split

Actually, the reason why [I grouped them](https://github.com/guardian/dotcom-rendering/pull/2587) was precisely because people _do_ use strange elements in MainMedia 🤷

Reply:
And yeah, importing css into ElementRenderer feels like a coupling we don't want to make - it trues creating a component is excessive for just this small amount of jsx code but the pattern is strong in this file and once you break it it's lost and then css will slowly spread everywhere.